pandas/tests/frame/methods/test_cov_corr.py

```diff
@@ -406,3 +409,86 @@ def test_corrwith_kendall(self):
         result = df.corrwith(df**2, method="kendall")
         expected = Series(np.ones(len(result)))
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "numeric_only, ser, expected",
+        [
+            (
+                True,
+                Series([0, 1, 1, 0]),
+                Series([0.0] * 3 + [1.0] * 4, index=list("ABCDEFH")),
+            ),
+            (
+                True,
+                Series([0.0, 1.0, 1.0, 0.0]),
+                Series([0.0] * 3 + [1.0] * 4, index=list("ABCDEFH")),
+            ),
+            (
+                True,
+                Series([False, True, True, False]),
+                Series([0.0] * 3 + [1.0] * 4, index=list("ABCDEFH")),
+            ),
+            (
+                False,
+                Series([0, 1, 1, 0]),
+                Series([0.0] * 3 + [1.0] * 5, index=list("ABCDEFGH")),
+            ),
+            (
+                False,
+                Series([0.0, 1.0, 1.0, 0.0]),
+                Series([0.0] * 3 + [1.0] * 5, index=list("ABCDEFGH")),
+            ),
+            (
+                False,
+                Series([False, True, True, False]),
+                Series([0.0] * 3 + [1.0] * 5, index=list("ABCDEFGH")),
+            ),
+            (
+                True,
+                Series([0, pd.NA, 1, 0], dtype="Int64"),
+                Series([0.0] * 3 + [1.0] * 4, index=list("ABCDEFH")),
+            ),
+            (
+                True,
+                Series([0.0, pd.NA, 1.0, 0.0], dtype="Float64"),
+                Series([0.0] * 3 + [1.0] * 4, index=list("ABCDEFH")),
```

Comment:
The parametrizations are basically the same except the length, please put this into the test. This is really hard to read like this

Reply:
remove parametrizations now